keanu-project/src/main/java/io/improbable/keanu/network/ProtobufReader.java

```diff
@@ -0,0 +1,236 @@
+package io.improbable.keanu.network;
+
+import com.google.common.primitives.Booleans;
+import com.google.common.primitives.Doubles;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import io.improbable.keanu.KeanuSavedBayesNet;
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.LoadParentVertex;
+import io.improbable.keanu.vertices.LoadVertexValue;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Parameter;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ProtobufReader implements NetworkReader {
+
+    private final Map<Vertex, KeanuSavedBayesNet.VertexValue> savedValues;
+
+    public ProtobufReader() {
+        savedValues = new HashMap<>();
+    }
+
+    @Override
+    public void loadValue(Vertex vertex) {
+        throw new IllegalArgumentException("Cannot Load value for Untyped Vertex");
```

Comment:
Can we ever load this? We could check for the vertex type first right?

Reply:
Yeah, we do check for other "loadable" types, and this is just the fallback and we wouldn't know what to do with it (we could just avoid saving anything, but that feels dirty).